### PR TITLE
Feature/link through browse list

### PR DESCRIPTION
**Pull request by TGrosskopf**
add a generic link item dialog with a data list view and paging

example implementation of the generic dialog in algorithm-publication-list to link existing publications
extension of data list to allow linking of elements

### CODE REVIEW

File: src/app/components/algorithms/algorithm-publications-list/algorithm-publications-list.component.ts [hunk 1]
```diff
@@ -104,34 +112,70 @@ export class AlgorithmPublicationsListComponent implements OnInit {
...
-    if (publicationData) {
-      // Search algorithms and filter only those that are not already linked
-      for (const publication of publicationData.publications) {
-        if (
-          !this.linkedPublications.some((publ) => publ.id === publication.id)
-        ) {
-          this.linkObject.data.push(publication);
+  openLinkPublicationDialog() {
+    this.publicationService.getPublications().subscribe((data) => {
+      this.updatePublicationData(JSON.parse(JSON.stringify(data)));
```
Comment (LHommeDeBat) [inline src/app/components/algorithms/algorithm-publications-list/algorithm-publications-list.component.ts:129]: Use types if possible here

File: src/app/components/algorithms/algorithm-publications-list/algorithm-publications-list.component.ts [hunk 2]
```diff
@@ -104,34 +112,70 @@ export class AlgorithmPublicationsListComponent implements OnInit {
...
+          noButtonText: 'Cancel',
+        },
+      });
+      const searchTextSub = dialogRef.componentInstance.onDataListConfigChanged.subscribe(
+        (search: QueryParams) => {
+          this.publicationService
+            .getPublications(search)
+            .subscribe((updatedData) => {
+              this.updatePublicationData(
+                JSON.parse(JSON.stringify(updatedData))
```
Comment (LHommeDeBat) [inline src/app/components/algorithms/algorithm-publications-list/algorithm-publications-list.component.ts:148]: Use Types here if possible

File: src/app/components/execution-environments/cloud-services/cloud-service-compute-resource-list/cloud-service-compute-resource-list.component.ts [hunk 3]
```diff
@@ -65,26 +83,69 @@ export class CloudServiceComputeResourceListComponent implements OnInit {
...
-          !this.linkedComputeResources.includes(computeResource)
-      );
-    } else {
-      this.linkObject.data = [];
-    }
+  openLinkComputeResourceDialog(): void {
+    this.executionEnvironmentsService
+      .getComputeResources()
+      .subscribe((data) => {
+        this.updateComputeResourcesData(JSON.parse(JSON.stringify(data)));
```
Comment (LHommeDeBat) [inline src/app/components/execution-environments/cloud-services/cloud-service-compute-resource-list/cloud-service-compute-resource-list.component.ts:98]: Use Types here if possible

File: src/app/components/execution-environments/cloud-services/cloud-service-compute-resource-list/cloud-service-compute-resource-list.component.ts [hunk 4]
```diff
@@ -65,26 +83,69 @@ export class CloudServiceComputeResourceListComponent implements OnInit {
...
+            noButtonText: 'Cancel',
+          },
+        });
+        const searchTextSub = dialogRef.componentInstance.onDataListConfigChanged.subscribe(
+          (search: QueryParams) => {
+            this.executionEnvironmentsService
+              .getComputeResources(search)
+              .subscribe((updatedData) => {
+                this.updateComputeResourcesData(
+                  JSON.parse(JSON.stringify(updatedData))
```
Comment (LHommeDeBat) [inline src/app/components/execution-environments/cloud-services/cloud-service-compute-resource-list/cloud-service-compute-resource-list.component.ts:117]: Use Types if possible

File: src/app/components/execution-environments/cloud-services/cloud-service-software-platform-list/cloud-service-software-platform-list.component.ts [hunk 5]
```diff
@@ -66,26 +84,69 @@ export class CloudServiceSoftwarePlatformListComponent implements OnInit {
...
+            noButtonText: 'Cancel',
+          },
+        });
+        const searchTextSub = dialogRef.componentInstance.onDataListConfigChanged.subscribe(
+          (search: QueryParams) => {
+            this.executionEnvironmentsService
+              .getSoftwarePlatforms(search)
+              .subscribe((updatedData) => {
+                this.updateSoftwarePlatformData(
+                  JSON.parse(JSON.stringify(updatedData))
```
Comment (LHommeDeBat) [inline src/app/components/execution-environments/cloud-services/cloud-service-software-platform-list/cloud-service-software-platform-list.component.ts:118]: Use Types here if possible

File: src/app/components/execution-environments/compute-resource/compute-resource-cloud-service-list/compute-resource-cloud-service-list.component.ts [hunk 6]
```diff
@@ -65,25 +83,66 @@ export class ComputeResourceCloudServiceListComponent implements OnInit {
...
+          noButtonText: 'Cancel',
+        },
+      });
+      const searchTextSub = dialogRef.componentInstance.onDataListConfigChanged.subscribe(
+        (search: QueryParams) => {
+          this.executionEnvironmentsService
+            .getCloudServices(search)
+            .subscribe((updatedData) => {
+              this.updateCloudServicesData(
+                JSON.parse(JSON.stringify(updatedData))
```
Comment (LHommeDeBat) [inline src/app/components/execution-environments/compute-resource/compute-resource-cloud-service-list/compute-resource-cloud-service-list.component.ts:114]: Use Types here if possible

File: src/app/components/execution-environments/compute-resource/compute-resource-software-platform-list/compute-resource-software-platform-list.component.ts [hunk 7]
```diff
@@ -68,26 +86,69 @@ export class ComputeResourceSoftwarePlatformListComponent implements OnInit {
...
+            noButtonText: 'Cancel',
+          },
+        });
+        const searchTextSub = dialogRef.componentInstance.onDataListConfigChanged.subscribe(
+          (search: QueryParams) => {
+            this.executionEnvironmentsService
+              .getSoftwarePlatforms(search)
+              .subscribe((updatedData) => {
+                this.updateSoftwarePlatformData(
+                  JSON.parse(JSON.stringify(updatedData))
```
Comment (LHommeDeBat) [inline src/app/components/execution-environments/compute-resource/compute-resource-software-platform-list/compute-resource-software-platform-list.component.ts:120]: Use Types here if possible

File: src/app/components/execution-environments/software-platforms/software-platform-impl-list/software-platform-impl-list.component.ts [hunk 8]
```diff
@@ -71,26 +90,67 @@ export class SoftwarePlatformImplListComponent implements OnInit {
...
+          noButtonText: 'Cancel',
+        },
+      });
+      const searchTextSub = dialogRef.componentInstance.onDataListConfigChanged.subscribe(
+        (search: QueryParams) => {
+          this.implementationService
+            .getImplementations(search)
+            .subscribe((updatedData) => {
+              this.updateImplementationData(
+                JSON.parse(JSON.stringify(updatedData))
```
Comment (LHommeDeBat) [inline src/app/components/execution-environments/software-platforms/software-platform-impl-list/software-platform-impl-list.component.ts:122]: Use Types here if possible

File: src/app/components/publications/publication-algorithms-list/publication-algorithms-list.component.ts [hunk 9]
```diff
@@ -48,37 +62,75 @@ export class PublicationAlgorithmsListComponent implements OnInit {
...
+          paginatorConfig: this.paginatorConfig,
+          noButtonText: 'Cancel',
+        },
+      });
+      const searchTextSub = dialogRef.componentInstance.onDataListConfigChanged.subscribe(
+        (search: QueryParams) => {
+          this.algorithmService
+            .getAlgorithms(search)
+            .subscribe((updatedData) => {
+              this.updateAlgorithmData(JSON.parse(JSON.stringify(updatedData)));
```
Comment (LHommeDeBat) [inline src/app/components/publications/publication-algorithms-list/publication-algorithms-list.component.ts:103]: Use Types here if possible

**Review by LHommeDeBat — APPROVED** [review]

**Review by ValeSayfa — APPROVED** [review]

**Pull request merged**